IPython/core/oinspect.py

```diff
@@ -1016,3 +1016,47 @@ def psearch(self,pattern,ns_table,ns_search=[],
             search_result.update(tmp_res)
 
         page.page('\n'.join(sorted(search_result)))
+
+
+def render_signature(obj, oname):
+    """
+    This was mostly taken from inspect.Signature.__str__.
+    Look there for the comments.
+    The only change is to add linebreaks when this gets too long.
+    """
+    if not isinstance(obj, inspect.Signature):
+        return oname + str(obj)
+
+    result = []
+    pos_only = False
+    kw_only = True
+    for param in self.parameters.values():
```

Comment:
I assume `self` here should be the signature object

Reply:
yup, whoops.